src/border_agent/border_agent.cpp

```diff
@@ -210,6 +215,68 @@ void BorderAgent::Stop(void)
     UnpublishMeshCopService();
 }
 
+#if OTBR_ENABLE_EPSKC
+void BorderAgent::HandleEpskcStateChanged(void *aContext)
+{
+    BorderAgent *borderAgent = static_cast<BorderAgent *>(aContext);
+    if (otBorderAgentIsEphemeralKeyActive(borderAgent->mNcp.GetInstance()))
+    {
+        borderAgent->PublishEpskcService();
+    }
+    else
+    {
+        borderAgent->UnpublishEpskcService();
+    }
+}
+
+void BorderAgent::PublishEpskcService()
+{
+    otInstance              *instance = mNcp.GetInstance();
+    int                      port     = otBorderAgentGetUdpPort(instance);
+    Mdns::Publisher::TxtData txtData;
+
+    otbrLogInfo("Publish meshcop-e service %s.%s.local. port %d", mServiceInstanceName.c_str(),
+                kBorderAgentEpskcServiceType, port);
+
+    mPublisher.PublishService(/* aHostName */ "", mServiceInstanceName, kBorderAgentEpskcServiceType,
+                              Mdns::Publisher::SubTypeList{}, port, txtData, [this](otbrError aError) {
+                                  if (aError == OTBR_ERROR_ABORTED)
+                                  {
+                                      // OTBR_ERROR_ABORTED is thrown when an ongoing service registration is
+                                      // cancelled. This can happen when the meshcop-e service is being updated
+                                      // frequently. To avoid false alarms, it should not be logged like a real error.
+                                      otbrLogInfo("Cancelled previous publishing meshcop-e service %s.%s.local",
+                                                  mServiceInstanceName.c_str(), kBorderAgentEpskcServiceType);
+                                  }
+                                  else
+                                  {
+                                      otbrLogResult(aError, "Result of publish meshcop-e service %s.%s.local",
+                                                    mServiceInstanceName.c_str(), kBorderAgentEpskcServiceType);
+                                  }
+
+                                  if (aError == OTBR_ERROR_DUPLICATED)
+                                  {
+                                      // Try to unpublish current service in case we are trying to register
+                                      // multiple new services simultaneously when the original service name
+                                      // is conflicted.
+                                      UnpublishEpskcService();
+                                      mServiceInstanceName = GetAlternativeServiceInstanceName();
```

Comment:
What you're doing here could potentially make the instance names of `_meshcop._udp` and `_meshcop-e._udp` different (at a very very low probability).

In the spec it says:
```
The <Instance> section of the DNS-SD Service Instance Name SHALL be chosen the same as the <Instance> section of the MeshCoP service.
```
 It's just a `SHALL` though. So just FYI.

Reply:
I'm fine with keeping the code as-is in this PR.

Some ideas for better handling this in a future PR:
1. There's some common code shared in publishing meshcop and meshcop-e. Maybe we can make a callback function used by both services.
2. In the callback function, when there's a name conflict, we can call something like `UpdateEpskcService` and `UpdateMeshCopService` at the same time.

Reply:
Thanks for these comments and suggestions. As we discussed, the possibility is kind of low and the modification may potentially impact the original Mesh-cop service; So we plan to keep it and keep close monitoring. Will trigger another fix if there is any.

Reply:
Add a comment to mark down the risk.